espflash/src/connection/mod.rs

```diff
@@ -174,18 +173,19 @@ impl Connection {
                 )));
             }
 
-            let read_slice = from_utf8(&buff[..read_bytes as usize]).map_err(|err| {
-                debug!("Error: {}", err);
-                Error::InvalidSerialRead
-            })?;
+            let read_slice = String::from_utf8_lossy(&buff[..read_bytes as usize]).into_owned();
```

Comment:
I was thinking of checking if the `read_slice` only contained the [REPLACEMENT_CHARACTER](https://doc.rust-lang.org/std/char/constant.REPLACEMENT_CHARACTER.html), but maybe it would be better to see if the `read_slice` contains `boot`:

```suggestion
            let read_slice = String::from_utf8_lossy(&buff[..read_bytes as usize]).into_owned();
             //Check if the read_slice contains other characters than REPLACEMENT_CHARACTER
            if !read_slice.contains("boot") {
                return Err(Error::InvalidSerialRead);
            }
```